spec/puppetserver/ca/generate_action_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'spec_helper'
+require 'utils/ssl'
+require 'shared_examples/cli_parsing'
+
+require 'puppetserver/ca/cli'
+require 'puppetserver/ca/generate_action'
+require 'puppetserver/ca/logger'
+require 'puppetserver/utils/signing_digest'
+require 'puppetserver/ca/host'
```

Comment:
I really don't like pulling in this many things here, but I think it will get significantly cleaner once we have a real CA object. I'm thinking we should either rename `Host` or fold it in elsewhere when we do that refactor; the fact that you can create multiple keys and CSRs with the same `Host` object seems misleading.